### PR TITLE
goasitop v0.1.2 (new formula)

### PR DESCRIPTION
**Pull request by metaspartan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/goasitop.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# typed: true
+# frozen_string_literal: true
+
+# goasitop homebrew formula
```
Comment (SMillerDev) [inline Formula/g/goasitop.rb:4]: This is not needed
```suggestion
```

File: Formula/g/goasitop.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class Goasitop < Formula
+  desc "Apple Silicon Monitor Top written in Go Lang"
+  homepage "https://github.com/context-labs/goasitop"
+  url "https://github.com/context-labs/goasitop/archive/refs/tags/v0.0.7.tar.gz"
+  sha256 "a17fb6b7ba87cd055371aa82abef7388cb6e8114cad92479f82eb58f5d9c46dc"
+  license "MIT"
+  depends_on "go" => :build
```
Comment (SMillerDev) [inline Formula/g/goasitop.rb:11]:
```suggestion
  license "MIT"

  depends_on "go" => :build
```

File: Formula/g/goasitop.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on arch: :arm64
+  depends_on :macos
+
+  # Restrict formula to macOS on Apple Silicon only
+
+  def install
+    system "go", "build", "-o", "#{bin}/goasitop"
```
Comment (SMillerDev) [inline Formula/g/goasitop.rb:18]:
```suggestion
    system "go", "build", *std_go_args
```

File: Formula/g/goasitop.rb (hunk 1 above)
Comment (metaspartan) [reply Formula/g/goasitop.rb:4]: It complains about missing the description in comments?

File: Formula/g/goasitop.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  test do
+    system "#{bin}/goasitop", "--version"
```
[the remark this answers is not in the document]
Comment (metaspartan) [reply Formula/g/goasitop.rb:23]: This is an in terminal monitor for Apple Silicon devices, it does create a log file we can test against. Will up the fixes soon.

File: Formula/g/goasitop.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/g/goasitop.rb:4]: I'm very sure it will not, since no formula in homebrew/core has this

File: Formula/g/goasitop.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  homepage "https://github.com/context-labs/goasitop"
+  url "https://github.com/context-labs/goasitop/archive/refs/tags/v0.1.0.tar.gz"
+  sha256 "e5e2211eb506725a4832a5b5096e2e6a61d3bb7ce0ab58674b50b1ec79187b5d"
+  license "MIT"
+
+  depends_on "go" => :build
+  depends_on arch: :arm64
+  depends_on :macos
+
+  # Restrict formula to macOS on Apple Silicon only
```
Comment (SMillerDev) [inline Formula/g/goasitop.rb:12]:
```suggestion
  # Restrict formula to macOS on Apple Silicon only
  depends_on arch: :arm64
  depends_on :macos
```

**Comment by SMillerDev** [comment]
Hi, this seems like a nice software, but not notable enough for us to package in Homebrew for now. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by metaspartan** [comment]
Yea already have the manual tap available in our repo. The software has not been advertised out yet (brand new), I will remake a PR when we reach the star, fork, and watcher requirements. Thanks.

**Pull request closed without merging**